### PR TITLE
#712 restructure processing

### PR DESCRIPTION
**Pull request by rozhkovdmitrii**

### CODE REVIEW

File: proxy/mempool/mem_pool.py [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
-        self.debug(f"Transaction is being processed on the worker: {eth_trx_hash}")
-
-    def do_extras(self):
-        pass
-
-    def __getstate__(self):
-        self_dict = self.__dict__.copy()
-        del self_dict['_pool']
-        return self_dict
+        self._pool = None
```
Comment (rozhkovdmitrii) [inline proxy/mempool/mem_pool.py]: delete this raw

File: proxy/mempool/mem_pool.py [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
+        self._tx_queue = asyncio.Queue()
+        self._pool = ProcessPoolExecutor(self.POOL_PROC_COUNT)
+        self._neon_tx_futures = set()
```
Comment (rozhkovdmitrii) [inline proxy/mempool/mem_pool.py]: git rid of it yet

File: proxy/mempool/mem_pool.py [hunk 3]
```diff
@@ -1,35 +1,40 @@
...
+        self._event_loop = asyncio.get_event_loop()
+        self._event_loop.create_task(self.do_mempool_stuff())
+        self._send_tx_futures = list()
```
Comment (rozhkovdmitrii) [inline proxy/mempool/mem_pool.py]: get rid of it yet

File: proxy/mempool/mempool_service.py [hunk 4]
```diff
@@ -1,26 +1,34 @@
...
+    def __init__(self):
+        self.event_loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.event_loop)
+        self._mempool_server = None
+        self._mempool = None
+        self._process = Process(target=self.run)
 
-    # QueueBasedService abstracts
+    def start(self):
+        self.info("Run until complete")
```
Comment (rozhkovdmitrii) [inline proxy/mempool/mempool_service.py:25]: I think this can be an extra logging

File: proxy/mempool/pickable_data_client.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import socket
+import pickle
+import struct
+from logged_groups import logged_group
+from typing import Any
```
Comment (rozhkovdmitrii) [inline proxy/mempool/pickable_data_client.py]: Move it a row up

**Review by afalaleev — APPROVED** [review]

**Pull request merged**